Kassiopeia/Interactions/Source/KSIntCalculatorMott.cxx

```diff
@@ -0,0 +1,175 @@
+#include "KSIntCalculatorMott.h"
+
+#include "KRandom.h"
+using katrin::KRandom;
+
+#include "KConst.h"
+
+using KGeoBag::KThreeVector;
+
+namespace Kassiopeia
+{
+
+KSIntCalculatorMott::KSIntCalculatorMott() : fThetaMin(0.), fThetaMax(0.), fMDCS(nullptr), fMTCS(nullptr) {}
+KSIntCalculatorMott::KSIntCalculatorMott(const KSIntCalculatorMott& aCopy) :
+    KSComponent(aCopy),
+    fThetaMin(aCopy.fThetaMin),
+    fThetaMax(aCopy.fThetaMax),
+    fMDCS(nullptr),
+    fMTCS(nullptr)
+{}
+KSIntCalculatorMott* KSIntCalculatorMott::Clone() const
+{
+    return new KSIntCalculatorMott(*this);
+}
+KSIntCalculatorMott::~KSIntCalculatorMott() = default;
+
+void KSIntCalculatorMott::CalculateCrossSection(const KSParticle& anInitialParticle, double& aCrossSection)
+{
+
+    double tInitialKineticEnergy = anInitialParticle.GetKineticEnergy_eV();
+    InitializeMDCS(tInitialKineticEnergy);
+
+    double fCrossSection = (fMTCS->Eval(fThetaMax) - fMTCS->Eval(fThetaMin)) * pow(10, -30);
+
+    DeinitializeMDCS();
+
+    aCrossSection = fCrossSection;
+
+    return;
+}
+void KSIntCalculatorMott::ExecuteInteraction(const KSParticle& anInitialParticle, KSParticle& aFinalParticle,
+                                                 KSParticleQueue& /*aSecondaries*/)
+{
+    double tTime = anInitialParticle.GetTime();
+    double tInitialKineticEnergy = anInitialParticle.GetKineticEnergy_eV();
+    KThreeVector tPosition = anInitialParticle.GetPosition();
+    KThreeVector tMomentum = anInitialParticle.GetMomentum();
+
+    double tTheta = GetTheta(tInitialKineticEnergy);
+    double tLostKineticEnergy = GetEnergyLoss(tInitialKineticEnergy, tTheta);
+
+
+    double tPhi = KRandom::GetInstance().Uniform(0., 2. * katrin::KConst::Pi());
+    tMomentum.SetPolarAngle(tTheta);
+    tMomentum.SetAzimuthalAngle(tPhi);
+
+    aFinalParticle.SetTime(tTime);
+    aFinalParticle.SetPosition(tPosition);
+    aFinalParticle.SetMomentum(tMomentum);
+    aFinalParticle.SetKineticEnergy_eV(tInitialKineticEnergy - tLostKineticEnergy);
+    aFinalParticle.SetLabel(GetName());
+
+    fStepAngularChange = tTheta * 180. / katrin::KConst::Pi();
+
+    return;
+}
+
+double KSIntCalculatorMott::GetTheta(const double& anEnergy){
+
+    double tTheta;
+
+    InitializeMDCS(anEnergy);
+
+    tTheta = fMDCS->GetRandom(fThetaMin, fThetaMax);
+
+    DeinitializeMDCS();
+
+    return tTheta;
+
+}
+
+double KSIntCalculatorMott::GetEnergyLoss(const double& anEnergy, const double& aTheta){
+    double M, me, p, anELoss;
+
+    M = 4.0026 * katrin::KConst::AtomicMassUnit_eV(); //  eV/c^2
```

Comment:
I think it would be useful to make the target mass `M` a user parameter that can be changed in the XML file.

Reply:
This calculator only works for Helium, so I guess the name is a bit misleading. Maybe I can change the name to he_mott_calculator?

Reply:
That is a good point! Would you also rename the classes and filenames then?

Reply:
Yes, I will update class names and filenames all together after I make the appropriate changes into a sole c++ implementation. I have to use rejection sampling to sample from the cross section, but I have never done this before so it might take me quite a bit of time. I originally used the TF1 class to shortcut this step, but using c++ will improve the performance as the other comment says

Reply:
You added a configuration option here for `Ne` and `He`. This is however specific to <sup>20</sup>Ne and <sup>4</sup>He, right? Is your entire code specific to those isotopes?